pallets/tinlake-investor-pool/src/lib.rs

```diff
@@ -471,6 +487,70 @@ pub mod pallet {
 			})
 		}
 
+		#[pallet::weight(100)]
+		pub fn set_max_reserve(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			max_reserve: T::Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(
+				Self::has_role_in_pool(pool_id, PoolRole::LiquidityAdmin, &who),
+				Error::<T>::NoPermission
+			);
+
+			Pool::<T>::try_mutate(pool_id, |pool| -> DispatchResult {
+				let pool = pool.as_mut().ok_or(Error::<T>::NoSuchPool)?;
+				pool.max_reserve = max_reserve;
+				Self::deposit_event(Event::MaxReserveSet(pool_id));
+				Ok(())
+			})
+		}
+
+		#[pallet::weight(100)]
+		pub fn update_pool(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			min_epoch_time: u64,
+			max_nav_age: u64,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			ensure!(
+				Self::has_role_in_pool(pool_id, PoolRole::PoolAdmin, &who),
+				Error::<T>::NoPermission
+			);
+
+			Pool::<T>::try_mutate(pool_id, |pool| -> DispatchResult {
+				let pool = pool.as_mut().ok_or(Error::<T>::NoSuchPool)?;
+
+				pool.min_epoch_time = min_epoch_time;
+				pool.max_nav_age = max_nav_age;
+				Self::deposit_event(Event::PoolUpdated(pool_id));
```

Comment:
Do we want to make these Update events a little more expressive? 

Like:
`PoolUpdate(T::PoolId, Reserve, ...)` so that one can see the new values without querying the storage.

Reply:
I didn't add it here because these values can be trivially queried, and @vedhavyas made the point to me recently that if this is the case, then it's probably not worth the additional storage of the event parameters.

Reply:
As long as client can fetch the reserve in o(1) using PoolID, I would lean towards using events hinting and client fetching the changes from state. Also makes it easy on storage

Reply:
Ah yes. Good idea, when they already got the pool I'd.